source/common/network/listener_impl.h

```diff
@@ -33,8 +33,9 @@ class ListenerImpl : public Listener {
    * Accept/process a new connection with the given remote address.
    * @param fd supplies the new connection's fd.
    * @param remote_address supplies the remote address for the new connection.
+   * @param remote_address supplies the remote port for the new connection.
    */
-  virtual void newConnection(int fd, const std::string& remote_address);
+  virtual void newConnection(int fd, const std::string& remote_address, uint32_t remote_port);
```

Comment:
I would not change the signature of this function. I would pass a fully resolved tcp://<address>:port string as remote_address, per other comments you might also pass local_address depending on how you implement.

Also, important note, this PR is subtly changing remoteAddress() to return a tcp://<>:<> style string instead of just a raw string. I think this is correct, but that behavior is going to break the client SSL filter here so this code needs to be fixed:
https://github.com/lyft/envoy/blob/master/source/common/filter/auth/client_ssl.cc#L100

(Perhaps the IP list code should just be fixed to expect a tcp:// style string)